### ISSUE TITLE
Fine tuning my custom dataset on already trained yolo

### ISSUE THREAD

**Issue by usamatariq70** [ultralytics/yolov5#2498]
## ❔Question
I trained my yolov5 and I had 7000 images with 33000 annotations. Now I want to finetune my that model and I have almost 5000 images and annotations. 
Can you please guide me which layers should I freeze to finetune it and how?

**Comment by glenn-jocher** [comment]
@usamatariq70 to start training from a previous model simply pass it to the --weights argument:
```
python train.py --data custom.yaml --weights path/to/ckpt.pt
```

**Comment by glenn-jocher** [comment]
@usamatariq70 for freezing layers see Transfer Learning with Frozen Layers Tutorial:

## YOLOv5 Tutorials

* [Train Custom Data](https://docs.ultralytics.com/yolov5/tutorials/train_custom_data)&nbsp; 🚀 RECOMMENDED
* [Weights & Biases Logging](https://github.com/ultralytics/yolov5/issues/1289)&nbsp; 🌟 NEW
* [Multi-GPU Training](https://docs.ultralytics.com/yolov5/tutorials/multi_gpu_training)
* [PyTorch Hub](https://docs.ultralytics.com/yolov5/tutorials/pytorch_hub_model_loading)&nbsp; ⭐ NEW
* [ONNX and TorchScript Export](https://docs.ultralytics.com/yolov5/tutorials/model_export)
* [Test-Time Augmentation (TTA)](https://docs.ultralytics.com/yolov5/tutorials/test_time_augmentation)
* [Model Ensembling](https://docs.ultralytics.com/yolov5/tutorials/model_ensembling)
* [Model Pruning/Sparsity](https://docs.ultralytics.com/yolov5/tutorials/model_pruning_and_sparsity)
* [Hyperparameter Evolution](https://docs.ultralytics.com/yolov5/tutorials/hyperparameter_evolution)
* [Transfer Learning with Frozen Layers](https://docs.ultralytics.com/yolov5/tutorials/transfer_learning_with_frozen_layers)&nbsp; ⭐ NEW
* [TensorRT Deployment](https://github.com/wang-xinyu/tensorrtx)

**Comment by SISTMrL** [comment]
> @usamatariq70 to start training from a previous model simply pass it to the --weights argument:
> 
> ```
> python train.py --data custom.yaml --weights path/to/ckpt.pt
> ```

just this command, it doesn't need to modify learning rate or other parameters? thanks!!!! looking forward to your reply

**Comment by glenn-jocher** [comment]
@SISTMrL 👋 Hello! Thanks for asking about **resuming training**. YOLOv5 🚀 Learning Rate (LR) [schedulers](https://github.com/ultralytics/yolov5/blob/9f3a388cea66ca71b9bbe057929eed8fbc920a7e/train.py#L109-L142) follow predefined LR curves for the fixed number of `--epochs` defined at training start (`default=300`), and are designed to fall to a minimum LR on the final epoch for best training results. For this reason you can not modify the number of epochs once training has started.

<img width="850" alt="LR Curves" src="https://user-images.githubusercontent.com/26833433/119332195-8e113a80-bc88-11eb-8eb7-7494ff1d5c01.png">

If your training was **interrupted** for any reason you may continue where you left off using the `--resume` argument. If your training fully completed, you can start a **new** training from any model using the `--weights` argument. Examples:

## Resume Single-GPU

You may not change settings when resuming, and **no additional arguments** other than `--resume` should be passed, with an optional path to the checkpoint you'd like to resume from. If no checkpoint is passed the most recently updated `last.pt` in your `yolov5/` directory is automatically found and used:
```bash
python train.py --resume  # automatically find latest checkpoint (searches yolov5/ directory)
python train.py --resume path/to/last.pt  # specify resume checkpoint
```

## Resume Multi-GPU

[Multi-GPU DDP trainings](https://docs.ultralytics.com/yolov5/tutorials/multi_gpu_training) must be resumed with the same GPUs and DDP command, i.e. assuming 8 GPUs:
```bash
python -m torch.distributed.launch --nproc_per_node 8 train.py --resume  # resume latest checkpoint
python -m torch.distributed.launch --nproc_per_node 8 train.py --resume path/to/last.pt  # specify resume checkpoint
```

## Start from Pretrained

If you would like to **start training** from a fully trained model, use the `--weights` argument, not the `--resume` argument:
```bash
python train.py --weights path/to/best.pt  # start from pretrained model
```

Good luck and let us know if you have any other questions!

**Comment by dbkalaria** [comment]
> @usamatariq70 to start training from a previous model simply pass it to the --weights argument:
> 
> ```
> python train.py --data custom.yaml --weights path/to/ckpt.pt
> ```

@glenn-jocher  I've tried fine-tuning my custom model using this command but instead of fine-tuned model I get a newly trained model. 

scenario: I've trained a model on 3 classes with 2 classes containing around 2500 images and 3rd class with only 500 images due to lack of data. Later I've gathered 1700 more for class 3 and fine-tuned my model on only images of 3rd class and while evaluating the fine-tuned model, it only predicts the 3rd class and some false positives for other two classes (predicts on 3rd class).

Can you please tell me what went wrong?

**Comment by glenn-jocher** [comment]
@dhruv312 everything is correct. You must train on all classes you expect your model to predict in deployment.

**Comment by dbkalaria** [comment]
@glenn-jocher so in fine-tuning also I need to give all the classes right?

**Comment by glenn-jocher** [comment]
@dhruv312 yes, during training the model is updated on the training data and nothing else. Whatever it's learned in the past is not actively retained, in fact quite the opposite weight decay by itself will bring all weights to zero even with zero losses.

**Comment by lchunleo** [comment]
can i check when we do the training on the pretrained model, are the layers all frozen? checking on the training code, it seem to me that the layers are not unfrozen, pls correct me if am wrong. thank you

**Comment by glenn-jocher** [comment]
@lchunleo no layers are frozen by default, they are all trainable, but you can use --freeze when training to freeze part of the model. See **Transfer Learning with Frozen Layers** Tutorial for details.

## Tutorials

- [Train Custom Data](https://docs.ultralytics.com/yolov5/tutorials/train_custom_data)  🚀 RECOMMENDED
- [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/)  ☘️
  RECOMMENDED
- [ClearML Logging](https://docs.ultralytics.com/yolov5/tutorials/clearml_logging_integration) 🌟 NEW
- [Weights & Biases Logging](https://github.com/ultralytics/yolov5/issues/1289)
- [Roboflow for Datasets, Labeling, and Active Learning](https://docs.ultralytics.com/yolov5/tutorials/roboflow_datasets_integration)  🌟 NEW
- [Multi-GPU Training](https://docs.ultralytics.com/yolov5/tutorials/multi_gpu_training)
- [PyTorch Hub](https://docs.ultralytics.com/yolov5/tutorials/pytorch_hub_model_loading)  ⭐ NEW
- [TFLite, ONNX, CoreML, TensorRT Export](https://docs.ultralytics.com/yolov5/tutorials/model_export) 🚀
- [Test-Time Augmentation (TTA)](https://docs.ultralytics.com/yolov5/tutorials/test_time_augmentation)
- [Model Ensembling](https://docs.ultralytics.com/yolov5/tutorials/model_ensembling)
- [Model Pruning/Sparsity](https://docs.ultralytics.com/yolov5/tutorials/model_pruning_and_sparsity)
- [Hyperparameter Evolution](https://docs.ultralytics.com/yolov5/tutorials/hyperparameter_evolution)
- [Transfer Learning with Frozen Layers](https://docs.ultralytics.com/yolov5/tutorials/transfer_learning_with_frozen_layers)  ⭐ NEW
- [Architecture Summary](https://docs.ultralytics.com/yolov5/tutorials/architecture_description)  ⭐ NEW

Good luck 🍀 and let us know if you have any other questions!

**Comment by astronitya** [comment]
> ## ❔Question
> I trained my yolov5 and I had 7000 images with 33000 annotations. Now I want to finetune my that model and I have almost 5000 images and annotations. Can you please guide me which layers should I freeze to finetune it and how?
> 
> ## Additional context

I have a big doubt here; if we are taking a pre trained model, and training it for a custom dataset, that means we are fine tunning the model, not the training it from scratch, right?

**Comment by dbkalaria** [comment]
> > ## ❔Question
> 
> > I trained my yolov5 and I had 7000 images with 33000 annotations. Now I want to finetune my that model and I have almost 5000 images and annotations. Can you please guide me which layers should I freeze to finetune it and how?
> 
> > 
> 
> > ## Additional context
> 
> 
> 
> I have a big doubt here; if we are taking a pre trained model, and training it for a custom dataset, that means we are fine tunning the model, not the training it from scratch, right?

@Nitya476 In fine tuning we are adjusting the pre trained model weights on our custom dataset. In fine tuning we are not training from scratch.
